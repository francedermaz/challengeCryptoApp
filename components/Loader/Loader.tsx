import { Image, View } from "react-native";
import { stylesLoader } from "./styles";

const loadingGif = require("../../assets/loadingPicture.gif");

export default function Loader() {
  return (
    <View style={stylesLoader.container} testID="loader-container">
      <Image
        testID="loading-gif"
        source={loadingGif}
        style={[stylesLoader.loadingGif]}
        resizeMode="contain"
      />
    </View>
  );
}
