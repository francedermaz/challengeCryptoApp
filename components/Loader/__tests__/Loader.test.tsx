import React from "react";
import { render } from "@testing-library/react-native";
import Loader from "../Loader";

test("should render the Loader component with the loadingGif", () => {
  const { getByTestId } = render(<Loader />);

  const container = getByTestId("loader-container");
  const loadingGif = getByTestId("loading-gif");

  expect(container).toBeTruthy();
  expect(loadingGif).toBeTruthy();
  expect(loadingGif.props.source).toBeDefined();
});

test("should match the Loader component snapshot", () => {
  const { toJSON } = render(<Loader />);

  expect(toJSON()).toMatchSnapshot();
});
