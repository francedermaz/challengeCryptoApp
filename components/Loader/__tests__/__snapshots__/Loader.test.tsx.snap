// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Loader component snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000c0",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "marginTop": "70%",
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 10000,
    }
  }
  testID="loader-container"
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      [
        {
          "width": "40%",
        },
      ]
    }
    testID="loading-gif"
  />
</View>
`;
